components/02-molecules/banner/grand-hero/yds-grand-hero.twig

```diff
@@ -0,0 +1,58 @@
+{#
+ # Available Props:
+ # - grand_hero__content__background
+ # - grand_hero__overlay_variation: contained (default), full
+ #
+ # Available Variables:
+ # - grand_hero__heading
+ # - grand_hero__snippet
+ # - grand_hero__link__content
+ # - grand_hero__link__url
+ #
+ # Available Blocks:
+ # - grand_hero__media
+ #}
+
+{% set grand_hero__base_class = 'grand-hero-banner' %}
+{% set grand_hero__content__background = grand_hero__content__background|default('basic-black') %}
+{% set grand_hero__overlay_variation = grand_hero__overlay_variation|default('contained') %}
+
+{% set grand_hero__attributes = {
+  'data-grand-hero-overlay-variation': grand_hero__overlay_variation,
+  class: bem(grand_hero__base_class),
+} %}
+
+<div {{ add_attributes(grand_hero__attributes) }}>
+  <div {{ bem('content-wrapper', [], grand_hero__base_class) }}>
+    <div {{ bem('image', [], grand_hero__base_class) }}>
+      {% block grand_hero__image %}
+        {% include "@atoms/images/image/_responsive-image.twig" %}
+      {% endblock %}
+    </div>
+    <div {{ bem('content', [], grand_hero__base_class) }} data-component-theme={{ grand_hero__content__background }}>
+      <div {{ bem('content-inner', [], grand_hero__base_class) }}>
+        <div {{ bem('text', [], grand_hero__base_class) }}>
```

Comment:
Since all of the fields are optional, we should wrap the content wrapper in an `if` that checks to make sure at least one of them is provided so that we don't print empty divs.